chalk-ir/src/interner.rs

```diff
@@ -272,19 +295,33 @@ mod default {
             tls::with_current_program(|prog| Some(prog?.debug_alias(alias, fmt)))
         }
 
+        fn debug_ty(ty: &Ty<ChalkIr>, fmt: &mut fmt::Formatter<'_>) -> Option<fmt::Result> {
+            tls::with_current_program(|prog| Some(prog?.debug_ty(ty, fmt)))
+        }
+
+        fn debug_lifetime(
+            lifetime: &Lifetime<ChalkIr>,
+            fmt: &mut fmt::Formatter<'_>,
+        ) -> Option<fmt::Result> {
+            tls::with_current_program(|prog| Some(prog?.debug_lifetime(lifetime, fmt)))
```

Comment:
so to fix the test failures, we could adapt this like so

```rust
tls::with_current_program(|prog| Some(prog?.debug_lifetime(lifetime, fmt)))
    .or_else(|| XXX)
```

where `XXX` gets the `lifetime.data` and (since we know it is an `Arc`...) just invokes `Debug` on it manually.

Reply:
the other (maybe better) option would be to get the `Program` in scope, as that will result in cleaner debug output anyhow